File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/geocoding/FoursquareResultLocationAdapter.java
package com.skedgo.tripkit.ui.geocoding;

import com.skedgo.geocoding.agregator.GCFoursquareResultInterface;
import com.skedgo.tripkit.common.model.Location;
import com.skedgo.tripkit.ui.data.places.Place;

import java.util.List;

public final class FoursquareResultLocationAdapter implements GCFoursquareResultInterface, ResultLocationAdapter<Place.TripGoPOI> {
    private final GCFoursquareResultInterface resultInterface;
    private final Location location;

    public FoursquareResultLocationAdapter(Location location, GCFoursquareResultInterface resultInterface) {
        this.location = location;
        this.resultInterface = resultInterface;
    }

    @Override
    public Place.TripGoPOI getPlace() {
        return new Place.TripGoPOI(location);
    }

    @Override
    public boolean isVerified() {
        return resultInterface.isVerified();
    }

    @Override
    public List<String> getCategories() {
        return resultInterface.getCategories();
    }

    @Override
    public String getName() {
        return resultInterface.getName();
    }

    @Override
    public Double getLat() {
        return resultInterface.getLat();
    }

    @Override
    public Double getLng() {
        return resultInterface.getLng();
    }
}

File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/geocoding/FoursquareResultLocationAdapter.kt
package com.skedgo.tripkit.ui.geocoding

import com.skedgo.geocoding.agregator.GCFoursquareResultInterface
import com.skedgo.tripkit.common.model.location.Location
import com.skedgo.tripkit.ui.data.places.Place.TripGoPOI

class FoursquareResultLocationAdapter(
    private val location: Location,
    private val resultInterface: GCFoursquareResultInterface
) : GCFoursquareResultInterface, ResultLocationAdapter<TripGoPOI?> {
    override fun getPlace(): TripGoPOI {
        return TripGoPOI(location)
    }

    override val isVerified: Boolean
        get() = resultInterface.isVerified

    override val categories: List<String>
        get() = resultInterface.categories

    override val name: String
        get() = resultInterface.name

    override val lat: Double?
        get() = resultInterface.lat

    override val lng: Double?
        get() = resultInterface.lng
}